### PR TITLE
e2e tests: Claim Bounty

### PR DESCRIPTION
**Pull request by sachintom999**
**Changes**
- Refactored `BaseE2ETest` to be reused across all the tests which requires login. Inheriting `BaseE2ETest` will automatically creates a test user and login the test user into the application. Any such test only need to handle the functionality and checks after login .
- Because of the above, refactored `e2e/test_login.py`
- Added tests for Claim Bounty functionality

**Testing**
test_clain_bounty

### CODE REVIEW

File: e2e/base.py [hunk 1]
```diff
@@ -20,7 +22,38 @@ def tearDownClass(cls):
...
+        self.page = self.browser.new_page()
+        
+
+        self.user_one = baker.make(
+            "security.User",
+            username=self.username,
+            password=self.password,
+            first_name=first_name,
+            last_name=last_name,
+        )
```
Comment (endalkh) [inline e2e/base.py:56]: Could you please use existing fixtures or create a new one in `conftest.py`?

File: e2e/test_claim_bounty.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            priority="High",
+            created_by=self.person,
+        )
+        self.bounty = baker.make(
+            "product_management.Bounty",
+            challenge=self.challenge,
+            title="Pet Sitting Made Easy",
+            status="Available",
+            description="Information Systems & Networking (Python)",
+        )
```
Comment (endalkh) [inline e2e/test_claim_bounty.py:42]: Could you please use existing fixtures or create a new one in conftest.py?

**Review by endalkh — CHANGES_REQUESTED** [review]

**Comment by adrianmcphee** [comment]
@sachintom999 are the changes requested by @endalkh sufficiently clear?

**Comment by sachintom999** [comment]
> @sachintom999 are the changes requested by @endalkh sufficiently clear?

@adrianmcphee , yes it clear . I got occupied and didnt get a chance to address it . Will push the changes in a couple of days .

**Review by endalkh — CHANGES_REQUESTED** [review]

**Comment by sachintom999** [comment]
@endalkh, thanks for your review comments. Really appreciate it.
Before I make the changes as per your last comments, I am planning to get the test pass in the pipeline. 
It is passing in my local, but somehow failing in the pipeline . 
Error - The  assertion for `BountyClaim` record , on clicking `Claim Bounty` button is failing.
I tried increasing the delay between the click and assertion , to make sure enough time for the db operation (  `BountyClaim` record creation) to complete. 

If you have bandwidth, could you please try at your end as well  ?

**Comment by endalkh** [comment]
> @endalkh, thanks for your review comments. Really appreciate it. Before I make the changes as per your last comments, I am planning to get the test pass in the pipeline. It is passing in my local, but somehow failing in the pipeline . Error - The assertion for `BountyClaim` record , on clicking `Claim Bounty` button is failing. I tried increasing the delay between the click and assertion , to make sure enough time for the db operation ( `BountyClaim` record creation) to complete.
> 
> If you have bandwidth, could you please try at your end as well ?

Hey @sachintom999, Thank you, Sure! I'll check that.

**Comment by sachintom999** [comment]
@endalkh thanks for the input. I will check that and get back.

**Pull request merged**